File: ios/NavigationView/HybridMapboxNavigationViewManagerRegistry.swift
//
//  HybridNitroMapboxNavigationViewManagerRegistry.swift
//  MapboxNavigation
//
//  Created by Fabrizio Beccaceci on 17/10/24.
//

import Foundation
import NitroModules


public class HybridMapboxNavigationViewManagerRegistry: HybridMapboxNavigationViewManagerRegistrySpec {
    
  public var hybridContext = margelo.nitro.HybridContext()
  
  public var memorySize: Int {
    return getSizeOf(self)
  }
  
  public func createViewManager(nitroId: Double) throws -> any HybridMapboxNavigationViewManagerSpec {
    print("getting view from map")
  
    return HybridMapboxNavigationViewManager(withNitroId: nitroId)
  }
  
}

File: android/src/main/java/com/margelo/nitro/com/mapboxnavigation/HybridMapboxNavigationViewManagerRegistry.kt
package com.margelo.nitro.com.mapboxnavigation

import com.margelo.nitro.core.Promise

class HybridMapboxNavigationViewManagerRegistry: HybridMapboxNavigationViewManagerRegistrySpec() {

  override val memorySize: Long
    get() = 0L

  override fun createViewManager(nitroId: Double): HybridMapboxNavigationViewManagerSpec {
    return Test()
  }

}

class Test() : HybridMapboxNavigationViewManagerSpec() {

  override val memorySize: Long
    get() = 0L

}
